wordpress-mobile/WordPress-Android
# [Mockito] Fix Unit Tests Execution Above JDK 11

FYI: This PR resolves an issue similar to [this](https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/2540), by applying a fix similar to [this](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2559). PS: Thanks @wzieba ! ❤️ 🙇 🚀 

-----

As per the above, this issue happens when running unit tests above JDK 11 and, the reason for this failure is `ByteBuddy` from Mockito.

- Replacing `nhaarman` with `mockito-kotlin` fixed the issue.
- Bumping `mockito` to latest `4.8.1` is optional, but high recommended (our current version is `3.3.3`, a very outdated version).

This PR fixes failing unit tests execution on JDKs higher than 11 by replacing `nhaarman` with `mockito-kotlin` and updating `mockito` to latest stable. With this update `nhaarman` is effectively removed from the codebase in favor of `mockiot-kotlin`.

-----

Review instructions:

- This b5e383dfdd5289a711a6ae67bd6839d774fcc926 was mainly done automatically, so reviewing all the code and import changes is not absolutely necessary.
- Firstly, @oguzkocer and @wzieba I am adding you two as reviewers to this PR, and that is, because we were all involved in the `FluxC` suchlike PR too, so it should be pretty straightforward for us to test and merge this. In addition to that, I am adding @AjeshRPai here too in order for us to get a sign-off from someone from WPAndroid as well.

-----

To test:

- Firstly, make sure you can reproduce the issue. Install any JDK above 11 and update `JAVA_HOME` (if you use terminal) or settings in Android Studio (if you use AS). `Hint: Android Studio Eel has embedded JDK version 17`
- Checkout `trunk`.
- Run unit tests.
- Assert unit tests fail with something like: `Mockito cannot mock this class (...)`
- Checkout this branch: `test/fix-unit-tests-execution-above-jdk11`
- Run unit tests again.
- Assert unit tests pass.

-----

## Regression Notes
1. Potential unintended areas of impact

Test are not running on all possible combinations of available and used JDKs.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:
